packages/vm/test/api/EIPs/eip-7002.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { Block } from '@ethereumjs/block'
+import { Chain, Common, Hardfork } from '@ethereumjs/common'
+import { LegacyTransaction } from '@ethereumjs/tx'
+import {
+  Account,
+  Address,
+  bigIntToBytes,
+  bytesToHex,
+  concatBytes,
+  hexToBytes,
+  setLengthLeft,
+  zeros,
+} from '@ethereumjs/util'
+import { assert, describe, expect, it } from 'vitest'
+
+import { VM } from '../../../src/vm.js'
+import { setupVM } from '../utils.js'
+
+const pkey = hexToBytes(`0x${'20'.repeat(32)}`)
+const addr = Address.fromPrivateKey(pkey)
+
+const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Cancun, eips: [7685, 7002] })
+
+const deploymentTxData = {
+  nonce: BigInt(0),
+  gasLimit: BigInt('0x3d090'),
+  gasPrice: BigInt('0xe8d4a51000'),
+  data: hexToBytes(
+    '0x61049d5f5561013280600f5f395ff33373fffffffffffffffffffffffffffffffffffffffe146090573615156028575f545f5260205ff35b366038141561012e5760115f54600182026001905f5b5f82111560595781019083028483029004916001019190603e565b90939004341061012e57600154600101600155600354806003026004013381556001015f3581556001016020359055600101600355005b6003546002548082038060101160a4575060105b5f5b81811460dd5780604c02838201600302600401805490600101805490600101549160601b83528260140152906034015260010160a6565b910180921460ed579060025560f8565b90505f6002555f6003555b5f548061049d141561010757505f5b60015460028282011161011c5750505f610122565b01600290035b5f555f600155604c025ff35b5f5ffd'
+  ),
```

Comment:
Can this code block get some docs? 😜

Reply:
Yes I need to add docs, the tests are in a super "local" state since it does not what I want it to do so did not yet bother to clean that all up.

For reference: it is the "Deployment" code from the EIP https://eips.ethereum.org/EIPS/eip-7002#deployment

Reply:
Ah ok, so the requests itself are not yet done, right?

Reply:
The code itself *should* generate requests, however, there is some unspecified behavior in the EIP which prevents me from actually submitting a withdrawal. I will update by throwing out this "mocked system call" (a la 4788) and instead perform the actual system call :) This will enable the requests.

Reply:
Done! Tests now cover the requests also! :smile: